src/java.base/share/classes/java/lang/reflect/code/Value.java

```diff
@@ -139,6 +142,45 @@ public boolean isDominatedBy(Value dom) {
     }
 
 
+    /**
+     * Returns {@code true} if this value is dominated by the given values {@code doms}.
+     *
+     * @param doms the dominating value
+     * @return {@code true} if this value is dominated by the given value {@code dom}.
+     * @throws IllegalStateException if the declaring block is partially built
+     */
+    public boolean isDominatedBy(Set<? extends Value> doms) {
```

Comment:
This method tests if the given set of values represent a dominating set for the actual value. I hope the implementation matches the "Dominating sets in directed graphs" definition.

I've moved it here because there are already two uses of the method:
- BytecodeGenerator::isDefinitelyAssigned needs to test for the dominance of VarStoreOps set to decide if the VarOp can be deferred 
- SlotToVarTransformer::isDominatedByTheSameVar needs to test dominance of set of SlotOp predecessors to identify initial SlotStoreOps.

Reply:
I understand the motivation, and the documentation is appreciated, but for i now i still think it better to let it soak within the bytecode package. I don't think it is ready to have such prominence in the core API, maybe it never will. That's not imply it is not useful for your purposes, but i am skeptical at the moment of its broad use.

The existing `isDominatedBy` methods on code items have quite a precise meaning with regards to nested hierarchy and block dominance. This method is very different and merges the concept of is-dominated-by-any-value with a dominance-like frontier approach, where values are substituted for their blocks (presumably within the same body, otherwise it does not make much sense) and there is some implicit relation between the values in those blocks. Splitting it in two concepts may make more sense.

Reply:
It has exactly same meaning as `isDominatedBy`. Second part is focused on blocks only because individual dominance test in the first part covers the situations where any member of the tested set precedes the value in the same block (or in the entry block) - so it is individually dominant and the method returns true. If any member of the set is individually dominant, the whole set is dominant.

Reply:
I've moved it to BytecodeUtil.

Reply:
Thank you. For the record i think it currently stretches the definition of dominance. For the second case consider the blocks connected as follows:
```
    e
   /  \
  a   b
   \  /
    c
```  
Where `this` value is in block `c` and the given values are in block `a` and `b` respectively. It is stretching the definition to say that `a` and `b` collectively dominate `c`. (FWIW in this case the dominance frontiers of `a` and `b` are the same, the set consisting of `c` i.e., `a` does not strictly dominate `c`, but some predecessor of `c` does, in this case `a`, which dominates itself.)